diff --git a/Android.bp b/Android.bp
--- a/Android.bp
+++ b/Android.bp
@@ -329,6 +329,7 @@ java_sdk_library {
     visibility: [
         "//build/soong/java/core-libraries",
         "//external/wycheproof",
+        "//libcore:__subpackages__",
         // Visibility for prebuilt conscrypt-module-sdk from the prebuilt of
         // this module.
         // TODO(b/155921753): Restrict this when prebuilts are in their proper
@@ -365,6 +366,7 @@ java_sdk_library {
         "//build/soong/java/core-libraries",
         "//frameworks/base",
         "//frameworks/base/api",
+        "//libcore:__subpackages__",
         "//packages/modules/common/sdk",
         // TODO(b/165823103): Remove visiblity for IPsec once CorePlatformApi is available
         "//packages/modules/IPsec",
