diff --git a/mmodules/intracoreapi/Android.bp b/mmodules/intracoreapi/Android.bp
--- a/mmodules/intracoreapi/Android.bp
+++ b/mmodules/intracoreapi/Android.bp
@@ -68,7 +68,6 @@ java_sdk_library {
     droiddoc_options: [
         "--hide-annotation libcore.api.Hide",
         "--show-single-annotation libcore.api.IntraCoreApi",
-        "--skip-annotation-instance-methods=false",
     ],
     merge_inclusion_annotations_dirs: ["ojluni-annotated-mmodule-stubs"],
 
